### PR TITLE
Add lifecycle stages section

### PR DESCRIPTION
**Pull request by aaronwolen**
Initial draft of the section describing the various lifecycle labels used to annotate SOMA classes and methods. Feedback and suggestions are welcome.

Related to https://github.com/single-cell-data/TileDB-SOMA/issues/487 and [sc-23225].

### CODE REVIEW

**Review by johnkerl — APPROVED** [review]

File: abstract_specification.md [hunk 1]
```diff
@@ -18,6 +18,15 @@ SOMA is an abstract _API specification_, with the goal of enabling multiple conc
 
 This document is attempting to codify the abstract, language-neutral SOMA data model and functional operations. Other specifications will document specific language bindings and particular storage system implementations. Where the term `language-specific SOMA specification` is used below, it implies a specification of the SOMA API as it is presented in a given language or computing environment (e.g., the SOMA Python API), common across all storage engine implementations in that language.
 
+# Lifecycle Stages
+
+The SOMA API uses [RStudio's lifecycle stage model](https://lifecycle.r-lib.org) to indicate the maturity of its classes and methods. The lifecycle stages are:
+
+- `experimental`: Under active development and may undergo significant and breaking changes.
```
Comment (thetorpedodog) [inline abstract_specification.md:25]: Worth adding something like "If an interface is unmarked, it should be assumed to be experimental."

**Review by thetorpedodog — APPROVED** [review]
looks good conceptually, one suggestion

File: abstract_specification.md (hunk 1 above)
Comment (aaronwolen) [reply abstract_specification.md:25]: Great suggestion. I'll spell out that experimental is currently the default.

File: abstract_specification.md [hunk 2]
```diff
@@ -18,6 +18,15 @@ SOMA is an abstract _API specification_, with the goal of enabling multiple conc
 
 This document is attempting to codify the abstract, language-neutral SOMA data model and functional operations. Other specifications will document specific language bindings and particular storage system implementations. Where the term `language-specific SOMA specification` is used below, it implies a specification of the SOMA API as it is presented in a given language or computing environment (e.g., the SOMA Python API), common across all storage engine implementations in that language.
 
+# Lifecycle Stages
+
+The SOMA API uses [RStudio's lifecycle stage model](https://lifecycle.r-lib.org) to indicate the maturity of its classes and methods. The lifecycle stages are:
```
Comment (bkmartinjr) [inline abstract_specification.md:23]: Suggest:   ... of its classes, methods, parameters or interfaces.  

In other words, I think it needs to include method parameters, global state (if any), etc.

**Review by bkmartinjr — APPROVED** [review]
Other than small nits on where this can be used, I like it.  The tags seem like a good selection that covers the user-facing viewpoint.

**Comment by bkmartinjr** [comment]
@aaronwolen - can we land this?

**Comment by johnkerl** [comment]
@Shelnutt2 can you approve or opt out? We've been using a 'unanimous' rule for review in this spec repo (contrasted with "one accept suffices unless stated otherwise" in the impl repo)

**Review by Shelnutt2 — APPROVED** [review]

**Comment by johnkerl** [comment]
Thanks @Shelnutt2 ! :)

**Pull request merged**